src/Features/LanguageServer/Protocol/Handler/Diagnostics/DocumentPullDiagnosticHandler.cs

```diff
@@ -43,16 +43,40 @@ protected override VSInternalDiagnosticReport CreateReport(TextDocumentIdentifie
                 Supersedes = WorkspaceDiagnosticIdentifier,
             };
 
-        protected override VSInternalDiagnosticParams[]? GetPreviousResults(VSInternalDocumentDiagnosticsParams diagnosticsParams)
-            => new[] { diagnosticsParams };
+        protected override PreviousResult[]? GetPreviousResults(VSInternalDocumentDiagnosticsParams diagnosticsParams)
+        {
+            if (diagnosticsParams.PreviousResultId != null && diagnosticsParams.TextDocument != null)
+            {
+                return new[] { new PreviousResult(diagnosticsParams.PreviousResultId, diagnosticsParams.TextDocument) };
+            }
 
-        protected override IProgress<VSInternalDiagnosticReport[]>? GetProgress(VSInternalDocumentDiagnosticsParams diagnosticsParams)
-            => diagnosticsParams.PartialResultToken;
+            return null;
+        }
```

Comment:
i think it would be worthwhile to doc this.  particularly why if either of these are null we just give null back.

(also, would any combination fo values here reflect an *error* in the caller?